mbf_abstract_nav/src/abstract_planner_execution.cpp

```diff
@@ -261,47 +256,48 @@ void AbstractPlannerExecution::run()
         current_tolerance = tolerance_;
         ROS_INFO_STREAM("A new goal pose is available. Planning with the new goal pose and the tolerance: "
                         << current_tolerance);
-        exceeded = false;
         geometry_msgs::Point g = goal_.pose.position;
         ROS_INFO_STREAM("New goal pose: (" << g.x << ", " << g.y << ", " << g.z << ")");
       }
 
-      make_plan = !(success || exceeded) || has_new_start_ || has_new_goal_;
-
       // unlock goal
       goal_start_mtx_.unlock();
-      setState(PLANNING, false);
-      if (make_plan)
+      if (cancel_)
```

Comment:
moved the cancel up so we don't call the costly makePlan method, if the user has already cancelled us